### ISSUE TITLE
Missing git client dependency

### ISSUE THREAD

**Issue by rishiv5** [jenkinsci/datadog-plugin#410]
**Describe the bug**
Jenkins startup shows the following error when loading datadog plugin:

```
WARNING h.ExtensionFinder$GuiceFinder$SezpozModule#configure: Failed to load org.datadog.jenkins.plugins.datadog.listeners.DatadogSCMListener
java.lang.ClassNotFoundException: org.jenkinsci.plugins.gitclient.GitClient
...
Caused: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/gitclient/GitClient
```

Looking into it further, I noticed that the datadog plugin imports the git-client plugin [here](https://github.com/jenkinsci/datadog-plugin/blob/master/src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogSCMListener.java#L64) but does not explicitly define a dependency in the `pom.xml`. Digging further into the dependency chain, it seems that the datadog plugin relied on an implicit dependency to git-client plugin through the pipeline-model-definition plugin. However, in their [latest update](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/706/files#diff-7b85a52253b5e2b14e61c527aaa89dc2cda490136562c2d9cda2dc06bb0b09ce), pipeline-model-definition no longer includes git-client as a dependency.

Proposal: Explicitly add git-client as a dependency of datadog plugin to ensure that git-client is loaded correctly during startup.

`severity/normal`

**To Reproduce**
Check Jenkins spin up logs or plugin list when installing the latest versions of datadog and pipeline-model-definition.

**Expected behavior**
Jenkins should contain the git-client plugin (via dependency resolution) if datadog plugin is installed.

**Screenshots**

**Environment and Versions (please complete the following information):**
Jenkins LTS 2.440.x
Pipeline: Declarative plugin (pipeline-model-definition) version after merge of https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/706

**Comment by nikita-tkachenko-datadog** [comment]
The fix has been released in the plugin v6.0.3. Please try updating and let me know if the issue persists. Thank you!

**Comment by rishiv5** [comment]
Seems to be resolved!
